include/vara/Feature/FeatureModelWriter.h

```diff
@@ -0,0 +1,124 @@
+#ifndef VARA_FEATURE_FEATUREMODELWRITER_H
+#define VARA_FEATURE_FEATUREMODELWRITER_H
+
+#include "vara/Feature/FeatureModel.h"
+
+#include "libxml/tree.h"
+#include <libxml/xmlwriter.h>
+
+#include <memory>
+
+namespace vara::feature {
+
+//===----------------------------------------------------------------------===//
+//                               FeatureModelParser Class
+//===----------------------------------------------------------------------===//
+
+/// \brief Base class for parsers with different input formats.
+class FeatureModelWriter {
+protected:
+  explicit FeatureModelWriter() = default;
+
+public:
+  virtual ~FeatureModelWriter() = default;
+
+  /// Build \a FeatureModel after parsing. May return null if parsing or
+  /// building failed.
+  ///
+  /// \returns an instance of \a FeatureModel or \a nullptr
+  virtual int writeFeatureModel(string path) = 0;
+
+};
+
+//===----------------------------------------------------------------------===//
+//                               FeatureModelXmlParser Class
+//===----------------------------------------------------------------------===//
+
+/// \brief Parsers for feature models in XML.
+class FeatureModelXmlWriter : public FeatureModelWriter {
+public:
+  explicit FeatureModelXmlWriter(FeatureModel fm) : fm(std::move(fm)) {}
+
+  int writeFeatureModel(string path) override;
+  int writeFeatureModel(char **buf);
```

Comment:
Is the double indirection here really needed? Do you repointer the pointer pointing to char* at some point?

Reply:
Yes, I do repoint the pointer, however, clang tidy wants me to use a container or smart pointer. I will look into those.